### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by c0d3ster**
removing security vulnerabilities by updating and locking all dependencies
Switched to yarn

Steps to ensure compatibility:
- passing tests within repo
- passing tests in EOSJS repo with locally referenced eosjs-ecc lib
- pushed transact calls successfully on local chain

any other ideas/suggestions for compatibility testing is much appreciated; we want to ensure this works for everyone before merging

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ const ecc = require('eosjs-ecc')
 
 # Include
 
--   Install with: `npm install eosjs-ecc`
+-   Install with: `yarn add eosjs-ecc`
```
Comment (chris-allnutt) [inline README.md]: why this change?  Also below you tell them to use yarn . Can't a user use either yarn OR npm, also I thought we were sticking with yarn for OSS.

Comment (c0d3ster) [reply README.md]: Yes, a user can use either but we are making the change over to yarn in all of our documentation.
